cms/djangoapps/api/v1/serializers/discussion_settings.py

```diff
@@ -0,0 +1,68 @@
+""" Discussion settings serializers. """
+
+
+from rest_framework import serializers
+
+from models.settings.course_metadata import CourseMetadata
+from xmodule.modulestore.django import modulestore
+
+
+def get_settings_serializer(value_field_cls):
+    """Creates & returns a settings serializer class dynamicly."""
+
+    class SettingSerializer(serializers.Serializer):
+
+        help = serializers.CharField(read_only=True)
+        hide_on_enabled_publisher = serializers.BooleanField(read_only=True)
+        display_name = serializers.CharField(read_only=True)
+        deprecated = serializers.BooleanField(read_only=True)
+        value = value_field_cls
+
+    return SettingSerializer()
```

Comment:
While this is a cleaver way around this I'm beginning to think that we should just adopt a simpler key:value for these settings in the API. 

The reason being:
- The help while useful for the current studio, will be out of place in the new UI where these will be toggles, and not a true-false input. 
- I don't think hide_on_enabled_publisher should affect the new MFE, and particularly not the discussion settings. 
- The display name can be duplicated in the new MFE especially since it's a new custom UI.
- This is a new API so there is no need for deprecated settings. 

That just leaves the value, which we can use directly.

Reply:
Removed all of those, only key pair value used.